lib/ueberauth/strategy/cas/user.ex

```diff
@@ -96,10 +96,18 @@ defmodule Ueberauth.Strategy.CAS.User do
     |> cast_value
   end
 
-  defp get_attribute_name(node) do
+  defp get_attribute_name(node, opts) do
     node
     |> xpath(~x"local-name(.)"s)
-    |> Macro.underscore()
+    |> sanitize_attribute_name(opts)
+  end
+
+  defp sanitize_attribute_name(name, opts) do
+    if Keyword.get(opts, :sanitize_attribute_names, true) do
+      Macro.underscore(name)
+    else
+      name
+    end
```

Comment:
The big idea is to be able to skip the `Macro.underscore/1` call.

We're using `ueberauth_cas` probably not in the most usual way: dynamically building the strategy to inject different configuration values depending on what the user is trying to access, with tens different IdP integrations. When we're trying to figure out an issue with IdP admins, having different attribute names is a bit confusing, especially since we're adding our own mapping rules on top of it.

Maybe it would make more sense for us to use a lower-level API (typically using `Ueberauth.Strategy.CAS.API` directly), but it could feel weird to provide the option only at the lower-level.

I chose the `sanitize_attribute_names` option name to:

- Match the existing `sanitize_string` function
- Prevent double-negation (e.g. `:disable_default_mapping: false` would be harded to read)